src/assets/Keycloak/keycloak.md

```diff
@@ -0,0 +1,69 @@
+
+1. `Identity Broker:`
+   Identity Broker acts as a middleman, linking various service providers with different identity providers. It establishes trust with external identity providers to utilize their identities for accessing internal services offered by service providers.
+   Imagine a textile company, which has two Keycloak setups: one for customer and another for internal employees. If you want your employees to access external services without creating new accounts on customer Keycloak, you can use the customer Keycloak
+   as the mediator and the customer Keycloak as the source in this scenario.
+
+2. `User Federation:`
+   Keycloak provides the ability to integrate with existing LDAP or Active Directory servers. It also supports implementation of your own provider such as a relational database.
+
+3. `Standard Protocols:`
+   Keycloak is based on three standard protocols i.e., OpenID Connect, OAuth2.0, and SAML.
+
+4. `Password Policies:`
+   Keycloak contain different password policies i.e., HashAlgorithm, Hashing Iterations, Digits, Lowercase Characters, Regular Expression and so on.
+
+5. `Single-Sign On:`
+   Keycloak facilitates a seamless login experience where users only need to sign in once to access multiple applications, eliminating the need for repeated logins.
+   This not only streamlines user experience but also simplifies identity management for administrators.
+
+6. `Social Login:`
+   Keycloak has built-in support for Google, GitHub, and Facebook which help us to use social identity providers.
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. `Complex Server Deployment:` The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+2. `Lack of Comprehensive Support Documentation:` Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## SetUp Keycloak using docker
+1. Keycloak using docker image
+```bash
+docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:24.0.2 start-dev
+```
+
+2. Keycloak with postgres database using docker-compose
+   ![Setup Keycloak using docker compose](src/assets/keycloak/images/start-keycloak-with-docker-compose.png)
+
+## Why You Should Know Keycloak
+1. Open source
+2. Supports three different authentication protocols which give you the possibility to cover many applications with different security demands with a single tool.
+3. Provides web-based GUI which makes any configurations changes easier.
+4. Huge community support
+
+## When It May Not Be the Best Choice
```

Comment:
```suggestion
## Why it May Not Be the Best Choice?
```

Reply:
Depends on what you want to say... When you use 'why' it means that it is not the best choice. By using 'when' it means that in certain situations it is the best choice and in other situations it's maybe not a good choice...